src/policy/interface.h

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2014 The Bitcoin developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_POLICY_INTERFACE_H
+#define BITCOIN_POLICY_INTERFACE_H
+
+#include <string>
+
+class CCoinsViewCache;
+class CScript;
+class CTransaction;
+
+/**
+ * \class CPolicy
+ * Interface class for non-consensus-critical policy logic, like whether or not
+ * a transaction should be relayed and/or included in blocks created.
+ */
+class CPolicy
+{
+public:
+    virtual ~CPolicy() {};
+    virtual bool ApproveScript(const CScript& scriptPubKey) const { return true; };
+    /**
+     * Check for standard transaction types
+     * @return True if all outputs (scriptPubKeys) use only standard transaction forms
+     */
+    virtual bool ApproveTx(const CTransaction& tx, std::string& reason) const { return true; };
```

Comment:
There is one default behavior: accept everything that follows the consensus rules.
I can turn CPolicy back into a pure abstract class, though I'm afraid @sipa @laanwj @theuni or @gmaxwell may have issues with a pure abstract class because it may be worse for performance.
I'm fine with either of the two options (not with implementing anything "reasonable" in CPolicy): either CPolicy accepts everything or it is a pure interface.